File: 프로그래머스/lv2/42587. 프로세스/프로세스.cpp
#include <bits/stdc++.h>
using namespace std;
using pii = pair<int, int>;

int solution(vector<int> priorities, int location) {

	queue<pii> q;
	for(int i=0; i<priorities.size(); i++)
		q.push({priorities[i], i});

	sort(priorities.begin(), priorities.end(), greater<>());

	int ret = 1;
	int mx = 0;
	while(!q.empty()) {
		pii cur = q.front();
		q.pop();

		if(cur.first == priorities[mx]) {
			if(cur.second == location) 
				break;
			mx++;
			ret++;
		}

		else q.push(cur);
	}
	return ret;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    q = deque([(p, i) for i, p in enumerate(priorities)])
    priorities.sort(reverse=True)

    idx = 0
    while q:
        cur = q.popleft()

        if cur[0] == priorities[idx]:
            if cur[1] == location:
                break
            idx += 1
        else:
            q.append(cur)

    return idx + 1